# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: devtools/server/connectors/js-window-actor/DevToolsWorkerChild.jsm
# Commit: cd01a13c1a4f
# Full Hash: cd01a13c1a4f23945c857b920c02d711ed5a0ded
# Author: Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date: 2020-11-09 21:53:49
# Description:
#   Bug 1674115 - [devtools] Fix crash caused by calling WorkerDebugger.setDebuggerReady. r=ochameau.
#   
#   We were trying to resume the worker debugger in case we
#   couldn't connect to the worker. Since connectToWorker can
#   reject when the worker is closed, we might try to resume a
# ==============================================================================

diff -r d5bc5231563e -r cd01a13c1a4f devtools/server/connectors/js-window-actor/DevToolsWorkerChild.jsm
--- a/devtools/server/connectors/js-window-actor/DevToolsWorkerChild.jsm	Mon Nov 09 09:57:12 2020 +0000
+++ b/devtools/server/connectors/js-window-actor/DevToolsWorkerChild.jsm	Mon Nov 09 10:03:41 2020 +0000
@@ -340,7 +340,11 @@
     try {
       await onConnectToWorker;
     } catch (e) {
-      dbg.setDebuggerReady(true);
+      // onConnectToWorker can reject if the Worker Debugger is closed; so we only want to
+      // resume the debugger if it is not closed (otherwise it can cause crashes).
+      if (!dbg.isClosed) {
+        dbg.setDebuggerReady(true);
+      }
       return;
     }
 
